src/zenml/integrations/kubernetes/orchestrators/kube_utils.py

```diff
@@ -169,26 +174,29 @@ def get_pod(
 
 
 def wait_pod(
-    core_api_fn: Callable[[], k8s_client.CoreV1Api],
+    orchestrator: "KubernetesOrchestrator",
```

Comment:
I don't think we want a dependency in our general kubernetes utils on the orchestrator, these might be used in other places and should not even be aware of what an orchestrator is. Instead I would suggest passing in a function as before, but maybe one that returns a kubernetes client? We could then pass in `orchestrator.get_kube_client` and achieve the same result, correct?

Reply:
Makes sense! I can pass `get_kube_client` instead of the orchestrator and call it inside the `wait_pod` function. Let me implement and test if that works.